bundle/Command/DeleteOldCollectedInfoCommand.php

```diff
@@ -0,0 +1,206 @@
+                    new InputOption('period', 'p', InputOption::VALUE_REQUIRED, 'Attributes older that this period will be deleted.'),
+                    new InputOption('all', 'a', InputOption::VALUE_NONE, 'Delete all fields.'),
+                    new InputOption('neglect', 'nn', InputOption::VALUE_NONE, 'Do not ask for confirmation.'),
+                ],
+            ),
+        );
+
+        $this->addUsage('--content-id=123 --field-identifiers=title,name,last_name');
+        $this->addUsage('--info-collection-id=456 --field-identifiers=title,name,last_name');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if (null === $input->getOption('content-id')) {
+            $output->writeln('<error>                                       </error>');
+            $output->writeln('<error>     Missing content-id parameter.     </error>');
+            $output->writeln('<error>                                       </error>');
+
+            return $this->displayHelp($input, $output);
+        }
+
+        if (null === $input->getOption('field-identifiers') && !$input->getOption('all')) {
+            $output->writeln('<error>                                              </error>');
+            $output->writeln('<error>     Missing field-identifiers parameter.     </error>');
+            $output->writeln('<error>                                              </error>');
+
+            return $this->displayHelp($input, $output);
+        }
+
+        if (null === $input->getOption('period')) {
+            $output->writeln('<error>                                       </error>');
+            $output->writeln('<error>     Missing period parameter.         </error>');
+            $output->writeln('<error>                                       </error>');
+
+            return $this->displayHelp($input, $output);
+        }
+
+        $contentId = (int) $input->getOption('content-id');
+        $fields = $this->getFields($input);
+
+        $info = sprintf('Command will delete <info>%s</info> fields for content #%d', empty($fields) ? 'all' : implode(', ', $fields), $contentId);
+        $output->writeln($info);
+
+        if ($this->proceedWithAction($input, $output)) {
+            $output->write('<info>Running.... </info>');
+
+            $filterCriteria = new Value\Filter\FilterCriteria(
+                new ContentId($contentId, 0, PHP_INT_MAX),
+                new DateTimeImmutable('@0'),
+                $this->getDateFromPeriod(),
+            );
+            $collections = $this->infoCollection->filterCollections($filterCriteria);
+
+            $extractIdFunction = function($collection) {
+                return $collection->getId();
+            };
+
+            $collectionsIds = array_map($extractIdFunction, $collections->getCollections());
+            $filterCollections = new Value\Filter\Collections($contentId, $collectionsIds);
```

Comment:
Why are we filtering twice?

Reply:
No, actually there is only one filtering. Explained in previous comment to iherak. https://github.com/netgen/NetgenInformationCollectionBundle/pull/91#discussion_r1484000848